docs/backends/generic-postgresql.rst

```diff
@@ -109,3 +109,66 @@ This is the 4.3 schema. Please find `the 4.2 schema <https://github.com/PowerDNS
 
 .. literalinclude:: ../../modules/gpgsqlbackend/schema.pgsql.sql
    :language: SQL
+
+CockroachDB
+-----------
+
+`CockroachDB <https://www.cockroachlabs.com/docs/stable/architecture/overview.html>`__ is a highly available, resilient database that focuses on scaling and consistency. Specifically: it offers a PostgreSQL like database interface,
+which means that most tools that talk the PostgreSQL protocol can use it.
+
+A few changes are needed on top of the generic PostgreSQL settings. CockroachDB does not natively support the range operators that some PowerDNS database queries use,
+and care must be taken that table index columns do not exceed the internal maximum integer size that PowerDNS uses.
+
+Schema differences
+^^^^^^^^^^^^^^^^^^
+
+Given the normal pgsql schema, change the following:
+
+1. Add explicit SEQUENCEs for all SERIAL columns:
+
+.. code-block:: SQL
+
+  CREATE SEQUENCE domain_id MAXVALUE 2147483648;
+  CREATE SEQUENCE record_id MAXVALUE 2147483648;
+  CREATE SEQUENCE comment_id MAXVALUE 2147483648;
+  CREATE SEQUENCE meta_id MAXVALUE 2147483648;
+  CREATE SEQUENCE key_id MAXVALUE 2147483648;
+  CREATE SEQUENCE tsig_id MAXVALUE 2147483648;
+
+2. Change all SERIAL / BIGSERIAL columns to use the SEQUENCEs. For instance:
+
+.. code-block:: SQL
+
+  -- Before
+  CREATE TABLE domains (
+    id                    SERIAL PRIMARY KEY,
+    ...
+  }
+
+  -- After
+  CREATE TABLE domains (
+    id                    INT PRIMARY KEY DEFAULT nextval('domain_id') PRIMARY KEY,
+    ...
+  );
+
+
+3. Do not add the recordorder INDEX to the records table, the text_pattern_ops operator class is not supported.
```

Comment:
Maybe show the specific index you mean?

I also wonder if a simple script that modifies the schema is in order to make things convenient.

Reply:
I'd rather include the entire SQL dump in that case, which is way more descriptive.

Reply:
The downside of that is that it will need to be maintained alongside the normal postgresql schema which might be sub-optimal. Maybe @pieterlexis or @Habbie have an opinion on that?

Reply:
https://github.com/PowerDNS/pdns/issues/5375#issuecomment-644822573

Reply:
I clarified which index to skip in the text.